### PR TITLE
bug(pkg): Add reproduction case for `cat` not working due to wrong `PATH`

### PR DESCRIPTION
**Pull request by Leonidas-from-XIV**
I ran across this case in #8225 and thought I can extract that into a free-standing example so we can fix it and have a regression test.

### CODE REVIEW

**Comment by Leonidas-from-XIV** [comment]
I think we discussed a similar issue with @Alizter and @gridbugs before.

File: test/blackbox-tests/test-cases/pkg/path.t [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  $ mkdir dependency-source
+  $ cat >dune.lock/dependency.pkg <<EOF
+  > (source (copy $PWD/dependency-source))
+  > EOF
+
+And we have a package we want to build
+
+  $ mkdir test-source
+  $ cat >test-source/parade <<EOF
+  > (=ↀωↀ=)✧  ヾ(*ΦωΦ)ﾉ
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/pkg/path.t:19]: Let's stick to ascii if you don't mind

File: test/blackbox-tests/test-cases/pkg/path.t [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  $ cat >dune.lock/test.pkg <<EOF
+  > (source (copy $PWD/test-source))
+  > ; adding deps breaks cat
+  > (deps dependency)
+  > (build
+  >  (system "cat parade"))
+  > EOF
+  $ dune build .pkg/test/target/
+  File ".pkg/test/_unknown_", line 1, characters 0-0:
+  /usr/bin/sh: line 1: cat: command not found
```
Comment (gridbugs) [inline test/blackbox-tests/test-cases/pkg/path.t:40]: This line is different on my machine:
```diff
-  /usr/bin/sh: line 1: cat: command not found
+  /bin/sh: cat: command not found
```

File: test/blackbox-tests/test-cases/pkg/path.t
Comment (gridbugs) [inline test/blackbox-tests/test-cases/pkg/path.t:1]: Can you name this file something more specific. To run this test manually the command is `dune build @test/path` which is very vague and likely to collide with another future test in another directory.

**Review by gridbugs — CHANGES_REQUESTED** [review]

**Comment by gridbugs** [comment]
In `pkg_rules.ml` if the `build_env` function returns an env containing a `PATH` it clobbers the `PATH` variable pulled from the user's environment. I guess we need to start extending the variable adding additional paths instead.

File: test/blackbox-tests/test-cases/pkg/path.t (hunk 2 above)
Comment (rgrinberg) [reply test/blackbox-tests/test-cases/pkg/path.t:40]: Indeed, we can use `command` to make the test a little more resilient. Taken from stack overflow:

```
command -v cat >/dev/null 2>&1 || { echo "no cat" }
```

File: test/blackbox-tests/test-cases/pkg/path.t (hunk 1 above)
Comment (Leonidas-from-XIV) [reply test/blackbox-tests/test-cases/pkg/path.t]: Sure, done!

File: test/blackbox-tests/test-cases/pkg/path.t
Comment (Leonidas-from-XIV) [reply test/blackbox-tests/test-cases/pkg/path.t:1]: Sure. Now that I opened the PR I can just name it after the PR number, so people can refer back to this discussion if they're confused :+1: as is the case with a few of our existing tests

**Comment by Leonidas-from-XIV** [comment]
To conclude, for future readers: this issue has been fixed by #8333. Thanks @gridbugs!

**Pull request merged**